endoreg_db/models/media/video/video_file_meta/text_meta.py

```
@@ -34,8 +34,25 @@ def _update_text_metadata(
+            if success:
+                # Refresh state after frame extraction
+                state.refresh_from_db()
+                if state.frames_extracted:
+                    logger.info(f"Successfully extracted frames for video {video.uuid}")
+                else:
+                    # Force update the state if extraction was successful but state wasn't updated
+                    state.frames_extracted = True
+                    state.save(update_fields=['frames_extracted'])
+                    logger.info(f"Corrected frames_extracted state for video {video.uuid}")
+            else:
+                raise ValueError(f"Cannot update text metadata for video {video.uuid}: Frame extraction failed.")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)